### PR TITLE
Filters

### PR DESCRIPTION
**Pull request by cpacia**
Add query filter for state on GET sales, purchases, and cases.

### CODE REVIEW

File: api/jsonapi.go [hunk 1]
```diff
@@ -2501,7 +2501,21 @@ func (i *jsonAPIHandler) GETPurchases(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	offsetId := r.URL.Query().Get("offsetId")
-	purchases, err := i.node.Datastore.Purchases().GetAll(offsetId, l)
+	stateQuery := r.URL.Query().Get("state")
```
Comment (tyler-smith) [inline api/jsonapi.go]: I would put then in a function like `parseSearchTerms(q url.Values) (string, []pb.OrderState)` and reuse it in the other endpoint.

File: repo/db/purchases.go [hunk 2]
```diff
@@ -119,17 +120,55 @@ func (p *PurchasesDB) Delete(orderID string) error {
 	return nil
 }
 
-func (p *PurchasesDB) GetAll(offsetId string, limit int) ([]repo.Purchase, error) {
+func (p *PurchasesDB) GetAll(offsetId string, limit int, stateFilter []pb.OrderState, searchTerm string) ([]repo.Purchase, error) {
 	p.lock.RLock()
 	defer p.lock.RUnlock()
 
+	stateFilterClause := ""
```
Comment (tyler-smith) [inline repo/db/purchases.go]: IMO all this state/filter/query building stuff should be in a single function as well.

**Comment by rmisio** [comment]
Can the total count and listing slug also be added to the responses or was that something you wanted to handle separately?

**Comment by cpacia** [comment]
Yeah I can add them in

**Comment by rmisio** [comment]
@cpacia @tyler-smith Do you guys have an ETA on this one? I'll be ready for it either later today or Monday (monday morning would be fine if do-able on your end).

Some additions that are needed:
- total count, total query count and listing slug added to the responses
- add a sorting param to sort by date (`sortBy=date-asc` / `sortBy=date-desc`)?

**Comment by cpacia** [comment]
I'll try to finish it today

**Comment by rmisio** [comment]
This is still happening. Can it please be fixed in this PR?
https://github.com/OpenBazaar/openbazaar-go/issues/524

**Comment by hoffmabc** [comment]
Yes! 🤙🏼

**Pull request merged**